engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -221,6 +233,19 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescript
       log.info(s"$tag transitioning from $fromState to $toState.")
   }
 
+  private def handleRetryableFailure(jobKey: BackendJobDescriptorKey) = {
```

Comment:
This will probably involve reporting the failed event to a metadata reporting service or something later on. Currently, it simply increases the number of attempts for the JobKey and resets it's status as `NotStarted` (so that it's picked up by the startRunnableCalls for re-execution). In essence, the ExecutionStore, at any given time, will only have jobs with either `NotStarted`, `Starting` / `Running`, `Failed` or `Done` statuses. If the use is more complex than this, please do let me know.